### PR TITLE
Enable linalg-to-xsmm in tpp-run

### PR DESCRIPTION
**Pull request by chelini**
Enable `linalg-to-xsmm` in tpp.run and fix mapping issue for binary and unary.
For unary and binary operations we need to use the captured operand and not
the operands of the linalg generic.

### CODE REVIEW

File: test/Integration/tpp-run-xsmm-path.mlir [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  %added = call @add(%fill, %fill_1) : (tensor<2x2xf32>, tensor<2x2xf32>) -> tensor<2x2xf32>
+  %c0 = arith.constant 0 : index
+  %d1 = arith.constant -1.0 : f32
+ 
+  // CHECK: ( ( 3, 3 ), ( 3, 3 ) ) 
+  %v1 = vector.transfer_read %added[%c0, %c0], %d1 : tensor<2x2xf32>, vector<2x2xf32>
+  vector.print %v1 : vector<2x2xf32>
+  
+  bufferization.dealloc_tensor %arg0 : tensor<2x2xf32>
+  bufferization.dealloc_tensor %arg1 : tensor<2x2xf32>
```
Comment (adam-smnk) [inline test/Integration/tpp-run-xsmm-path.mlir:41]: Is this manual memory management really needed?

File: lib/TPP/ConvertLinalgToXsmm.cpp [hunk 2]
```diff
@@ -101,17 +102,16 @@ static void replaceOpWithUnary(RewriterBase &rewriter,
       loc, integer64, kind, dims, flags, dtype);
   SmallVector<Value> invokeOperands;
   invokeOperands.push_back(dispatched);
-  invokeOperands.append(linalgOp->getOperands().begin(),
-                        linalgOp->getOperands().end());
+  invokeOperands.append(operands.begin(), operands.end());
```
Comment (adam-smnk) [inline lib/TPP/ConvertLinalgToXsmm.cpp:105]: Does any test capture this change?

Comment (chelini) [reply lib/TPP/ConvertLinalgToXsmm.cpp:105]: yes, relu_5 and add_9. The add has 2 operands and the xsmm lowering has 3 operands. Same for the relu 1 operand and 2 operands in xsmm.

File: test/Integration/tpp-run-xsmm-path.mlir (hunk 1 above)
Comment (chelini) [reply test/Integration/tpp-run-xsmm-path.mlir:41]: I saw this pattern in other tests. I can remove it

Comment (adam-smnk) [reply test/Integration/tpp-run-xsmm-path.mlir]: It's only for weird edge cases in bufferization when there is (or at least used to be) issue with lifetimes.

File: test/Integration/tpp-run-xsmm-path.mlir [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      ^bb0(%in: f32, %in_1: f32, %out: f32):
+        %0 = arith.addf %in, %in_1 : f32
+        linalg.yield %0 : f32
+  } -> tensor<2x2xf32>
+  return %0 : tensor<2x2xf32>
+}
+
+func.func @entry() {
+  %cst_one = arith.constant 1.0 : f32
+  %arg0 = bufferization.alloc_tensor() : tensor<2x2xf32>
```
Comment (adam-smnk) [inline test/Integration/tpp-run-xsmm-path.mlir:25]: nit: use empty instead?

**Review by adam-smnk — APPROVED** [review]
